client/state/directly/actions.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * Internal dependencies
+ */
+import {
+	initialize as initializeDirectly,
+	askQuestion as askDirectlyQuestion,
+	maximize as maximizeDirectly,
+	minimize as minimizeDirectly,
+	openAskForm as openDirectlyAskForm,
+} from 'lib/directly';
+import {
+	DIRECTLY_ASK_QUESTION,
+	DIRECTLY_INITIALIZE,
+	DIRECTLY_MAXIMIZE,
+	DIRECTLY_MINIMIZE,
+	DIRECTLY_OPEN_ASK_FORM,
+} from 'state/action-types';
+
+export const askQuestion = ( questionText, name, email ) => ( dispatch ) => {
+	dispatch( { type: DIRECTLY_ASK_QUESTION, questionText, name, email } );
+	askDirectlyQuestion( questionText, name, email );
+};
```

Comment:
> I was advised to move them to Action Creators to avoid dispatching new actions within middleware.

I'm not sure where you got this information but unfortunately something got lost in translation. The middleware API is explicitly given access to the `dispatch()` method of the store _plus_ it's given `next()` as a way of passing an action along the middleware chain.

As @jordwest indicated _any_ side-effecting operation should be moved into the middleware and away from the actions themselves.

I'd be happy to discuss this further with you and share about the operation and role of middleware, if you would like.